p2p/connected_test.go

```diff
@@ -26,47 +29,71 @@ func TestAnnounceConnected(t *testing.T) {
 		nodes[i] = node.(*QriNode)
 	}
 
+	// set up bus & listener for `ETP2PQriPeerConnected` events
```

Comment:
I'm looking back on this test, and wondering what exactly it is supposed to be testing.

First, we set up a network of nodes & explicitly connect them.
Then we create a separate node that hasn't been connected explicitly, and only connect it to a single node.
Then we "AnnounceConnected" and wait until we get a message from each node.

Are we testing that AnnounceConnected works? Or are we trying to tell if we can get connections to each node just by connecting to one node in the network? 

This also ignores the part that sometimes nodes are connecting without us explicitly connecting them, but they do it through dhs, not because of a 2nd degree qri connection.

I'm wondering if this is doing too much, and we should be explicitly connecting the main test node to each node in the test network. The test would be much more reliable.

Reply:
> Are we testing that AnnounceConnected works? Or are we trying to tell if we can get connections to each node just by connecting to one node in the network?

Both. The definition of working for `AnnounceConnected` is to propagate that message to other qri peers.

> This also ignores the part that sometimes nodes are connecting without us explicitly connecting them, but they do it through dhs, not because of a 2nd degree qri connection.

I'm assuming you mean through the DHT here, and yes there are lots of ways qri peers can connect. This test is checking one of those ways.

Is this test causing problems? I'm happy to disable it for now if so. To me the bigger issues I'm seeing in CI is the resurfaced `TestConnect` failure in the cmd package, and making sure`TestQriIdentityService` passes consistently. Those two enough are a huge slog, so I'd be happy to disable this to focus on those

Reply:
re: dht, yea typo

I'm a bit confused about exactly how AnnounceConnected is supposed to fit in with everything.

From following the code right now, we have 
- `AnnounceConnected` which sends a message to each connected qri peer w/ "my" addr info. It is being used in one place: after we've had one successful connection to a bootstrap peer, we `AnnounceConnected`
- `handleConnected` checks to see if we've received this message before, if not, it decodes the message and requests the profile from the original node.

But this is just what `upgradeQriConnection` and now `QriIdentityService` does except from the other direction, we wait until the connection has been fully identified by libp2p and then we request the peer's profile. In this case, we are saying instead "Hey I'm here, request my profile", but to peers that we have already established on (our side) that are qri connections. Those peers don't turn around and further propagate our info to their peers.

If the goal is that we should be propagating the existence of newly connected peers, I think we should be doing it another way. 

Once we have "upgraded" to a qri connection, we should send a message to our other connections "hey this peer exists". There should be a loop in the `QriIdentityService` that listens for this messages. If you receive one of these messages, check if you are already connected the peer. If so, drop ignore it. If not, establish a connection, which will trigger the QriIdentityService, which will propagate that peer to your already connected peers.

I think this is what we are trying to achieve, and our code may have done that at one point, but we've changed so much that it's not currently doing that.

Reply:
Completely agree. Your most recent comment clarifies: this isn't so much about what the test is doing as: we have overlapping functionality between `AnnounceConnected` and the new identity service. Let's remove `AnnounceConnected`